frontend/passkeys.js

```diff
@@ -0,0 +1,35 @@
+document.addEventListener('DOMContentLoaded', function() {
+    'use strict';
+
+    const registerPasskeyButtons = document.querySelectorAll('#register-passkey');
+
+
+    async function registerPasskey() {
+        const response = await fetch('/passkey/registration-options') //fetch PublicKeyCredentialCreationOptions as JSON string
+        const publicKeyCredentialCreationOptionsJSON = await response.json() // convert to JSONObject
+        const credentialCreationOptions = PublicKeyCredential.parseCreationOptionsFromJSON(publicKeyCredentialCreationOptionsJSON); // convert to PublicKeyCredentialCreationOptions
+        const publicKeyCredential = await navigator.credentials.create({publicKey: credentialCreationOptions}); // create PublicKeyCredential
+        console.log('publicKeyCredential: ', publicKeyCredential);
+        const registrationResponseJSON = publicKeyCredential.toJSON(); // convert to JSONObject
```

Comment:
Not sure that the comments here are useful.  They were originally included as a guide because this whole block of code was copied from https://webauthn4j.github.io/webauthn4j/en/#generating-a-webauthn-credential-key-pair